### PR TITLE
Update airlift, drift, parquet, picocli

### PR DESCRIPTION
**Pull request by wendigo**
Supersedes https://github.com/trinodb/trino/pull/21802. Added @Fokko as a coauthor of the last commit.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -179,7 +179,7 @@
         <!-- keep dependency properties sorted -->
         <dep.accumulo-hadoop.version>2.7.7-1</dep.accumulo-hadoop.version>
         <dep.accumulo.version>3.0.0</dep.accumulo.version>
-        <dep.airlift.version>247</dep.airlift.version>
+        <dep.airlift.version>248</dep.airlift.version>
```
Comment (findepi) [inline pom.xml:182]: How come this works without updating discovery?

Comment (wendigo) [reply pom.xml:182]: Discovery was updated along with 247

Comment (wendigo) [reply pom.xml:182]: And this is 248

File: lib/trino-parquet/src/main/java/io/trino/parquet/ParquetMetadataConverter.java [hunk 2]
```diff
@@ -155,6 +156,7 @@ public static LogicalTypeAnnotation getLogicalTypeAnnotation(LogicalType type)
                 yield timestampType(timestamp.isAdjustedToUTC, convertTimeUnit(timestamp.unit));
             }
             case UUID -> uuidType();
+            case FLOAT16 -> float16Type();
```
Comment (findepi) [inline lib/trino-parquet/src/main/java/io/trino/parquet/ParquetMetadataConverter.java:159]: Should this be test covered?

cc @Fokko

File: plugin/trino-delta-lake/pom.xml [hunk 3]
```diff
@@ -167,6 +167,12 @@
         <dependency>
             <groupId>org.apache.parquet</groupId>
             <artifactId>parquet-format-structures</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>javax.annotation</groupId>
+                    <artifactId>javax.annotation-api</artifactId>
```
Comment (findepi) [inline plugin/trino-delta-lake/pom.xml:173]: Why? Is the exclusion avoidabe?

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
@martint @raunaqmorarka PTAL parquet changes

File: plugin/trino-delta-lake/pom.xml (hunk 3 above)
Comment (wendigo) [reply plugin/trino-delta-lake/pom.xml:173]: It's not. It's a transitive dependency used by Parquet. We are on jakarta and javax is a banned one.

File: lib/trino-parquet/src/main/java/io/trino/parquet/ParquetMetadataConverter.java (hunk 2 above)
Comment (wendigo) [reply lib/trino-parquet/src/main/java/io/trino/parquet/ParquetMetadataConverter.java]: Extracted to https://github.com/trinodb/trino/pull/21861

**Comment by Fokko** [comment]
Thanks @wendigo for picking this up 🎉

**Pull request merged**